packages/components/src/components/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useState, forwardRef, ForwardedRef } from "react";
+import map from "lodash/map";
+import filter from "lodash/filter";
+import includes from "lodash/includes";
+import toLower from "lodash/toLower";
+import isBoolean from "lodash/isBoolean";
+import isFunction from "lodash/isFunction";
+import {
+  ComboboxProps as AriakitComboboxProps,
+  useComboboxStore,
+} from "@ariakit/react";
+import find from "lodash/find";
+import { ComboboxInput } from "./ComboboxInput/ComboboxInput";
+import { ComboboxPopover } from "./ComboboxPopover/ComboboxPopover";
+import { ComboboxItem } from "./ComboboxItem/ComboboxItem";
+import { ComboboxProvider } from "./ComboboxProvider/ComboboxProvider";
+
+export type ComboboxItemProps = {
+  label: string;
+  value: string;
+  disabled?: boolean;
+};
+
+export type ComboboxProps<T extends ComboboxItemProps = ComboboxItemProps> =
+  Omit<AriakitComboboxProps, "store"> & {
+    /** The options for the combobox. */
+    items: T[];
+    /** The function that will be called when the input value changes. */
+    onChangeInput?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    /** The function that will be called when an option is selected. */
+    onSelectItem?: (value: string[] | string) => void;
+    /** The placeholder text for the input. */
+    placeholder?: string;
+    /** The message to display when there are no results. */
+    noResultsMessage?: string;
+    /**
+     * The function to filter the items based on the input value.
+     * If it is true, it will filter the items based on the input value.
+     * If it is a function, it will filter the items based on the input value and the item.
+     */
+    filterItem?: boolean | ((inputValue: string, item: T) => boolean);
+    /** Indicates if the combobox has an error. */
+    hasError?: boolean;
+    /** A custom store to be used instead of the default one. */
+    customStore?: AriakitComboboxProps["store"];
+  };
+
+const matchSorter = (
+  list: ComboboxItemProps[],
+  value: string,
+): ComboboxItemProps[] => {
+  return filter(list, (item) => includes(toLower(item.label), toLower(value)));
+};
+
+const getFilteredItems = <T extends ComboboxItemProps>(
+  items: T[],
+  searchValue: string,
+  filterItem?: boolean | ((inputValue: string, item: T) => boolean),
+): T[] => {
+  if (isBoolean(filterItem)) {
+    return filterItem
+      ? (matchSorter(items as ComboboxItemProps[], searchValue) as T[])
+      : items;
+  }
+  if (isFunction(filterItem)) {
+    return filter(items, (item) => filterItem(searchValue, item));
+  }
+  return items;
+};
+
+// eslint-disable-next-line jsdoc/require-param
```

Comment:
we should add `jsdoc/require-returns` here too

Reply:
Done [here](https://github.com/Localitos/pluto/pull/1788/files#diff-395d307cf2a5c77a478d9c4ed84231437e5ee35ebc7e642c119997cdafc3c7ffR72)